spyder/plugins/preferences/widgets/configdialog.py

```diff
@@ -0,0 +1,171 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright © Spyder Project Contributors
+# Licensed under the terms of the MIT License
+# (see spyder/__init__.py for details)
+
+from spyder.config.base import _, load_lang_conf
+from spyder.config.manager import CONF
+from spyder.utils import icon_manager as ima
+
+from qtpy.QtCore import QSize, Qt, Signal, Slot
+from qtpy.QtWidgets import (QDialog, QDialogButtonBox, QHBoxLayout,
+                            QListView, QListWidget, QListWidgetItem,
+                            QPushButton, QScrollArea, QSplitter,
+                            QStackedWidget, QVBoxLayout)
+
+class ConfigDialog(QDialog):
+    """Spyder configuration ('Preferences') dialog box"""
+
+    # Signals
+    check_settings = Signal()
+    size_change = Signal(QSize)
+
+    def __init__(self, parent=None):
+        QDialog.__init__(self, parent)
+
+        self.main = parent
```

Comment:
Is this really necessary? Widgets should not have a reference to the Main window, but instead emit signals.